### PR TITLE
Use std::unordered_map as hash_map when available

### PR DESCRIPTION
**Pull request by jarzec**
Resolves deprecation warnings with GCC 6.4 and compile errors with GCC 7.3

### CODE REVIEW

**Review by daviddoria — COMMENTED** [review]
What's the reason for removing the unused overload? Since this is a library, might someone want to use this function?

File: source/Hash.h [hunk 1]
```diff
@@ -33,6 +33,9 @@
 
 #ifndef HASH_INCLUDED
 #define HASH_INCLUDED
+
+#if __cplusplus < 201103L
```
Comment (daviddoria) [inline source/Hash.h:37]: Are there other c++11 versions besides March (03)? If not, I'd add a comment like "// c++11" or something to clarify. (I've never used these values, but I'd not heard it called 201103 before).

Comment (jarzec) [reply source/Hash.h:37]: I think 201103 is the date when the committee approved C++1 as finalized, that is why I chose that

File: source/MultiGridOctreeData.inl [hunk 2]
```diff
@@ -2653,14 +2653,6 @@ inline int Octree<Degree>::GetEdgeLoops(std::vector<std::pair<long long,long lon
 	return int(loops.size());
 }
 
-template<int Degree>
-inline int Octree<Degree>::AddTriangles(CoredMeshData* mesh,std::vector<CoredPointIndex> edges[3],std::vector<Point3D<float> >* interiorPositions,const int& offSet)
-{
-	std::vector<CoredPointIndex> e;
-	for(int i=0;i<3;i++){for(size_t j=0;j<edges[i].size();j++){e.push_back(edges[i][j]);}}
-	return AddTriangles(mesh,e,interiorPositions,offSet);
```
Comment (jarzec) [inline source/MultiGridOctreeData.inl:2661]: This call is incorrect - it lacks the 5th argument. With GCC up to version 6 this compiles as the function is never used. With GCC 7 this leads to compilation error. An alternative to removing the method could be adding a 5th parameter to be directly passed on here.

**Pull request merged**